File: C++/Kata.LoanPrediction.C++/Kata.LoanPrediction.C++/LoanCalculationOutput.cpp
//
//  LoanCalculationOutput.cpp
//  Kata.LoanPrediction.C++
//
//  Created by a.j.dowding on 3/09/2015.
//  Copyright (c) 2015 a.j.dowding. All rights reserved.
//

#include "LoanCalculationOutput.h"

/*
 ***************************************
 ** Default constructor.
 ***************************************
 */
LoanCalculationOutput::LoanCalculationOutput()
{
    this->interestStartDate = *DATETIME_EMPTY;
    this->loanEndsDate = *DATETIME_EMPTY;
    this->totalInterestPaid = 0.0;
    this->targetEndDateHit = false;
    this->targetEndDateMissedInDays = 0;
}

/*
 ***************************************
 ** Constructor with initial arguments.
 ***************************************
 */
LoanCalculationOutput::LoanCalculationOutput(dateTime interestStartDate, dateTime loanEndsDate, double totalInterestPaid, bool targetEndDateHit, int targetEndDateMissedInDays)
{
    this->interestStartDate = interestStartDate;
    this->loanEndsDate = loanEndsDate;
    this->totalInterestPaid = totalInterestPaid;
    this->targetEndDateHit = targetEndDateHit;
    this->targetEndDateMissedInDays = targetEndDateMissedInDays;
}

/*
 ***************************************
 ** Gets the interest start date.
 ***************************************
 */
dateTime LoanCalculationOutput::getInterestStartDate()
{
    return(interestStartDate);
}

/*
 ***************************************
 ** Set the interest start date.
 ***************************************
 */
void LoanCalculationOutput::setInterestStartDate(dateTime source)
{
    interestStartDate = source;
}

/*
 ***************************************
 ** Gets the loan end date.
 ***************************************
 */
dateTime LoanCalculationOutput::getLoanEndsDate()
{
    return(loanEndsDate);
}

/*
 ***************************************
 ** Sets the loan end date.
 ***************************************
 */
void LoanCalculationOutput::setLoanEndsDate(dateTime source)
{
    loanEndsDate = source;
}

/*
 ***************************************
 ** Gets the total interest paid.
 ***************************************
 */
double LoanCalculationOutput::getTotalInterestPaid()
{
    return(totalInterestPaid);
}

/*
 ***************************************
 ** Sets the total interest paid.
 ***************************************
 */
void LoanCalculationOutput::setTotalInterestPaid(double source)
{
    totalInterestPaid = source;
}

/*
 ***************************************
 ** Gets the target end date hit.
 ***************************************
 */
bool LoanCalculationOutput::getTargetEndDateHit()
{
    return(targetEndDateHit);
}

/*
 ***************************************
 ** Sets the target end date hit.
 ***************************************
 */
void LoanCalculationOutput::setTargetEndDateHit(bool source)
{
    targetEndDateHit = source;
}

/*
 ***************************************
 ** Gets the number of days the target
 ** end date missed.
 ***************************************
 */
int LoanCalculationOutput::getTargetEndDateMissedInDays()
{
    return(targetEndDateMissedInDays);
}

/*
 ***************************************
 ** Sets the number of days the target 
 ** end date missed.
 ***************************************
 */
void LoanCalculationOutput::setTargetEndDateMissedInDays(int source)
{
    targetEndDateMissedInDays = source;
}

/*
 ***************************************
 ** Adds a transaction.
 ***************************************
 */
void LoanCalculationOutput::addTransaction(LoanTransaction transaction)
{
    transactions.push_back(transaction);
}

/*
 ***************************************
 ** Gets a const pointer to a const 
 ** vector of transactions.
 ***************************************
 */
const vector<LoanTransaction>* const LoanCalculationOutput::getTransactionList()
{
    return(&transactions);
}

File: C#/Kata.LoanPrediction.CSharp/Kata.LoanPrediction.CSharp.Common/Models/LoanCalculationOutput.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kata.LoanPrediction.CSharp.Common.Models
{
    /// <summary>
    /// Represents the output from the
    /// LoanCalculator.
    /// </summary>
    public class LoanCalculationOutput
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="LoanCalculationOutput"/> class.
        /// </summary>
        public LoanCalculationOutput()
        {
            Transactions = new List<LoanTransaction>();
        }

        /// <summary>
        /// Gets the transactions.
        /// </summary>
        /// <value>
        /// The transactions.
        /// </value>
        public List<LoanTransaction> Transactions { get; private set; }

        /// <summary>
        /// Gets or sets the interest start date.
        /// </summary>
        /// <value>
        /// The interest start date.
        /// </value>
        public DateTime InterestStartDate { get; set; }

        /// <summary>
        /// Gets or sets the loan ends date.
        /// </summary>
        /// <value>
        /// The loan ends date.
        /// </value>
        public DateTime LoanEndsDate { get; set; }

        /// <summary>
        /// Gets or sets the total interest paid.
        /// </summary>
        /// <value>
        /// The total interest paid.
        /// </value>
        public double TotalInterestPaid { get; set; }

        /// <summary>
        /// Gets or sets a value indicating whether [target end date hit].
        /// </summary>
        /// <value>
        ///   <c>true</c> if [target end date hit]; otherwise, <c>false</c>.
        /// </value>
        public bool TargetEndDateHit { get; set; }

        /// <summary>
        /// Gets or sets the target end date missed in days.
        /// </summary>
        /// <value>
        /// The target end date missed in days.
        /// </value>
        public int TargetEndDateMissedInDays { get; set; }
    }
}
